### PR TITLE
ledger-tool: Adjust the help messages for force-update-to-open

### PR DESCRIPTION
**Pull request by steviez**
#### Problem
The previous help message was pretty generic as it covered two potential error scenarios. 

#### Summary of Changes
This change gives a more detailed error message to each of those errors.

I tested this out with some manual ledger-tool runs to ensure that the error catching still works as before, but with new messages displayed for each case (missing blockstore and missing column)

### CODE REVIEW

File: ledger-tool/src/ledger_utils.rs [hunk 1]
```diff
@@ -365,20 +365,36 @@ pub fn open_blockstore(
...
+                    "Failed to open blockstore at {ledger_path:?}, it \
+                    does not have all necessary columns: {err:?}"
+                );
+            } else {
+                eprintln!("Failed to open blockstore at {ledger_path:?}: {err:?}");
+                exit(1);
+            }
             if !force_update_to_open {
-                error!("Use --force-update-to-open to allow blockstore to update");
+                eprintln!("Use --force-update-to-open flag to attempt to update the blockstore");
```
Comment (taozhu-chicago) [inline ledger-tool/src/ledger_utils.rs:397]: to me `error!()` is better, for I `grep ERROR` and that'd tell me what went wrong. `eprintln!()` assumes I know what I am looking for.

**Review by taozhu-chicago — COMMENTED** [review]
thanks for clarifying the errors! Just a comment about logging

File: ledger-tool/src/ledger_utils.rs (hunk 1 above)
Comment (steviez) [reply ledger-tool/src/ledger_utils.rs:397]: Yeah, I'm inbetween on this.
- I personally like / am used to seeing `ERROR` ... the red text on terminal output and having `ERROR` to grep for are convenient
- I heard some feedback that the log statements look more "imposing" to a casual user whereas the regular `println!() / eprintln!()` are not.

But, I think that the feedback about logs is probably more pertinent to to users of `solana` CLI; `solana-ledger-tool` is primarily a tool used by people working on/with validator client, and as such, the logs are probably more familiar. Philosophically, I think `solana-ledger-tool` is more similar to `solana-validator` than it is to the CLI.

That being said, the rest of the `ledger-tool` currently uses `eprintln!()`, so the switch to `eprintln!()` makes these consistent. For the sake of this PR, I'd lean towards consistency, and then maybe in a subsequent PR, I can bump all the `eprintln!()` (or at least all the ones that make sense) to appropriate log (`error!()` / `warn!()` / etc). Thoughts ?

Comment (taozhu-chicago) [reply ledger-tool/src/ledger_utils.rs:397]: consistency is always appreciated, plus 

>  it exits immediately after these lines 

makes good argument. 😄

**Review by taozhu-chicago — APPROVED** [review]
lgtm

**Pull request merged**